barretenberg/cpp/src/barretenberg/dsl/acir_proofs/honk_contract.hpp

```diff
@@ -292,34 +292,45 @@ struct Transcript {
     Fr lookupGrandProductDelta;
 }
 
-library TranscriptLib
-{
-    function generateTranscript(Honk.Proof memory proof,
-                                Honk.VerificationKey memory vk,
-                                bytes32[] calldata publicInputs) internal view returns(Transcript memory t)
+library TranscriptLib {
+    function generateTranscript(Honk.Proof memory proof, bytes32[] calldata publicInputs, uint256 publicInputsSize)
+        internal
+        view
+        returns (Transcript memory t)
     {
-        (t.eta, t.etaTwo, t.etaThree) = generateEtaChallenge(proof, publicInputs);
+        Fr previousChallenge;
+        (t.eta, t.etaTwo, t.etaThree, previousChallenge) = generateEtaChallenge(proof, publicInputs, publicInputsSize);
 
-        (t.beta, t.gamma) = generateBetaAndGammaChallenges(t.etaThree, proof);
+        (t.beta, t.gamma, previousChallenge) = generateBetaAndGammaChallenges(previousChallenge, proof);
 
-        t.alphas = generateAlphaChallenges(t.gamma, proof);
+        (t.alphas, previousChallenge) = generateAlphaChallenges(previousChallenge, proof);
 
-        t.gateChallenges = generateGateChallenges(t.alphas[NUMBER_OF_ALPHAS - 1]);
+        (t.gateChallenges, previousChallenge) = generateGateChallenges(previousChallenge);
 
-        t.sumCheckUChallenges = generateSumcheckChallenges(proof, t.gateChallenges[CONST_PROOF_SIZE_LOG_N - 1]);
-        t.rho = generateRhoChallenge(proof, t.sumCheckUChallenges[CONST_PROOF_SIZE_LOG_N - 1]);
+        (t.sumCheckUChallenges, previousChallenge) = generateSumcheckChallenges(proof, previousChallenge);
+        (t.rho, previousChallenge) = generateRhoChallenge(proof, previousChallenge);
 
-        t.zmY = generateZMYChallenge(t.rho, proof);
+        (t.zmY, previousChallenge) = generateZMYChallenge(previousChallenge, proof);
 
-        (t.zmX, t.zmZ) = generateZMXZChallenges(t.zmY, proof);
+        (t.zmX, t.zmZ, previousChallenge) = generateZMXZChallenges(previousChallenge, proof);
 
         return t;
     }
 
-    function generateEtaChallenge(Honk.Proof memory proof, bytes32[] calldata publicInputs)
-        internal view returns(Fr eta, Fr etaTwo, Fr etaThree)
+    function splitChallenge(Fr challenge) internal pure returns (Fr first, Fr second) {
+        uint256 challengeU256 = uint256(Fr.unwrap(challenge));
+        uint256 lo = challengeU256 & 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+        uint256 hi = challengeU256 >> 128;
+        first = FrLib.fromBytes32(bytes32(lo));
+        second = FrLib.fromBytes32(bytes32(hi));
+    }
+
+    function generateEtaChallenge(Honk.Proof memory proof, bytes32[] calldata publicInputs, uint256 publicInputsSize)
```

Comment:
publicInputsSize doesn't seem needed here as an input

Reply:
but maybe this was needed to maintain the same interface as Transcript.sol? unsure, didn't want to test it so I'll leave it